src/power.jl

```diff
@@ -71,29 +61,50 @@ function simulate_waldtests(
     nβ, m = length(β), deepcopy(morig)
     # we need to do for in-place operations to work across threads
     m_threads = [m]
+    coef_threads = [similar(β)]
+    stderr_threads = [similar(β)]
+    zstat_threads = [similar(β)]
+    pval_threads = [similar(β)]
+    iter_threads = [similar(β, Int)]
 
     if use_threads
         Threads.resize_nthreads!(m_threads)
+        Threads.resize_nthreads!(coef_threads)
+        Threads.resize_nthreads!(stderr_threads)
+        Threads.resize_nthreads!(zstat_threads)
+        Threads.resize_nthreads!(pval_threads)
+        Threads.resize_nthreads!(iter_threads)        
     end
 
+    χsq = Chisq(1)
+
+    cnms = Symbol.(coefnames(morig))
+
+    tidynms = (:coefname,:iteration,:β,:se,:z,:p)
+    results = sizehint!(NamedTuple{tidynms,Tuple{Symbol,Int,T,T,T,T}}[], n)
+
     rnglock = ReentrantLock()
+    pushlock = ReentrantLock()
+    iter = 0
     replicate(n, use_threads=use_threads) do
         mod = m_threads[Threads.threadid()]
         lock(rnglock)
+        iter += 1
+        fill!(iter_threads[Threads.threadid()], iter)
```

Comment:
This line can be moved outside the lock -- only the incrementing and RNG needs to be within this lock.